rmgpy/molecule/molecule.py

```diff
@@ -1593,12 +1593,12 @@ def generate_H_bonded_structures(self):
         structs = []
         Hbonds = self.find_H_bonds()
         for i,bd1 in enumerate(Hbonds):
-            molc = deepcopy(self)
+            molc = self.copy(deep=True)
```

Comment:
If you pass a Molecule object to copy.deepcopy, all of the edges in the graph get duplicated because it copies all of the attributes of each atom (and there are always two atoms that have the same bond object as an attribute), but is not smart enough to know that it has already created a copy of the bonds. To overcome this, Molecule objects have a method that behaves like you think deepcopy would but without duplicating the number of bonds. Without this change, enumerating the bonds will double count everything if hydrogen bonded structures are generated.